keyvault/client/src/main/java/com/azure/keyvault/models/DeletedSecret.java

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.keyvault.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+
+public class DeletedSecret extends SecretAttributes {
+
+    /**
+     * The url of the recovery object, used to identify and recover the deleted
+     * secret.
+     */
+    @JsonProperty(value = "recoveryId")
+    private String recoveryId;
+
+    /**
+     * The time when the secret is scheduled to be purged, in UTC.
+     */
+    private OffsetDateTime scheduledPurgeDate;
+
+    /**
+     * The time when the secret was deleted, in UTC.
+     */
+    private OffsetDateTime deletedDate;
+
+    /**
+     * Get the recoveryId value.
+     *
+     * @return the recoveryId value
+     */
+    public String recoveryId() {
+        return this.recoveryId;
+    }
+
+    /**
+     * Set the recoveryId value.
+     *
+     * @param recoveryId the recoveryId value to set
+     * @return the DeletedSecret object itself.
+     */
+    public DeletedSecret recoveryId(String recoveryId) {
+        this.recoveryId = recoveryId;
+        return this;
+    }
+
+    /**
+     * Get the scheduledPurgeDate value.
+     *
+     * @return the scheduledPurgeDate value
+     */
+    public OffsetDateTime scheduledPurgeDate() {
+        if (this.scheduledPurgeDate == null) {
+            return null;
```

Comment:
@JonathanGiles What is the way to return a default OffsetDateTime value? Is `null` the answer? Does this trigger Checkstyle errors? Or would Optional<T> be the way? Or... MIN/MAX?

Reply:
Optional is ok, but can be verbose. I don't like returning null much either. Min/Max isn't a good choice obviously. I haven't decided on guidance yet about whether we should use Optional or not. Keen to hear thoughts...

Reply:
The Deleted Secret in KV only exists if the vault is soft-delete enabled.
Once the secret gets deleted, the scheduled purge date is auto generated for it.
It is always provided by Azure KV response. I can't think of a case, where this value will be a null response from Azure KV.
But, thinking about it generally, Optional is one possible approach to have here.